corporate/lib/stripe.py

```diff
@@ -1276,49 +1278,65 @@ def create_card_update_session(self) -> Dict[str, Any]:
-        customer.default_discount = new_discount
+        old_monthly_discounted_price = customer.monthly_discounted_price
+        customer.monthly_discounted_price = monthly_discounted_price
+        old_annual_discounted_price = customer.annual_discounted_price
+        customer.annual_discounted_price = annual_discounted_price
+        # Ideally we would have some way to restore flat discounted months
+        # if we applied discounted to a customer and reverted it but seems
+        # like an edge case and can be handled manually on request.
         customer.flat_discounted_months = 0
-        customer.save(update_fields=["default_discount", "flat_discounted_months"])
+        customer.save(
+            update_fields=[
+                "monthly_discounted_price",
+                "annual_discounted_price",
+                "flat_discounted_months",
+            ]
+        )
         plan = get_current_plan_by_customer(customer)
         if plan is not None and plan.tier == customer.required_plan_tier:
-            self.apply_discount_to_plan(plan, new_discount)
+            self.apply_discount_to_plan(plan, customer)
 
         # If the customer has a next plan, apply discount to that plan as well.
         # Make this a check on CustomerPlan.SWITCH_PLAN_TIER_AT_PLAN_END status
         # if we support this for other plans.
         next_plan = self.get_legacy_remote_server_next_plan(customer)
         if next_plan is not None and next_plan.tier == customer.required_plan_tier:
-            self.apply_discount_to_plan(next_plan, new_discount)
+            self.apply_discount_to_plan(next_plan, customer)
 
         self.write_to_audit_log(
             event_type=AuditLogEventType.DISCOUNT_CHANGED,
             event_time=timezone_now(),
-            extra_data={"old_discount": old_discount, "new_discount": new_discount},
-        )
-        new_discount_string = (
-            format_discount_percentage(new_discount) if (new_discount != Decimal(0)) else "0"
-        )
-        old_discount_string = (
-            format_discount_percentage(old_discount)
-            if (old_discount is not None and old_discount != Decimal(0))
-            else "0"
+            extra_data={
+                "old_monthly_discounted_price": old_monthly_discounted_price,
+                "new_monthly_discounted_price": customer.monthly_discounted_price,
+                "old_annual_discounted_price": old_annual_discounted_price,
+                "new_annual_discounted_price": customer.annual_discounted_price,
+            },
         )
-        return f"Discount for {self.billing_entity_display_name} changed to {new_discount_string}% from {old_discount_string}%."
+
+        return f"Monthly price for {self.billing_entity_display_name} changed to {customer.monthly_discounted_price} from {old_monthly_discounted_price}. Annual price changed to {customer.annual_discounted_price} from {old_annual_discounted_price}."
```

Comment:
@alya yeah, this is the new text for confirmation banner.

Reply:
Cool, thanks!